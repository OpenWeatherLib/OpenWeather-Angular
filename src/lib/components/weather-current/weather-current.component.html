<mat-progress-bar *ngIf="(isLoading$ | async)" mode="indeterminate"></mat-progress-bar>
<mat-card class="ga-card" *ngIf="(weatherCurrent$ | async) && !(isLoading$ | async)">
  <mat-card-header>
    <img mat-card-avatar class="ga-card-header-image" src="{{(weatherCurrent$ | async)?.weatherCondition?.icon}}" alt="Weather current icon">
    <mat-card-title>Current Weather</mat-card-title>
    <mat-card-subtitle>{{(weatherCurrent$ | async)?.weatherCondition?.description}}</mat-card-subtitle>
  </mat-card-header>
  <img mat-card-image src="{{(weatherCurrent$ | async)?.weatherCondition?.wallpaper}}" alt="Weather current wallpaper">
  <mat-card-content>
    <p>Temperature: {{(weatherCurrent$ | async)?.main?.temp}}°C
      ({{(weatherCurrent$ | async)?.main?.temp_min}}°C-{{(weatherCurrent$ | async)?.main?.temp_max}}°C)</p>
    <p>Pressure: {{(weatherCurrent$ | async)?.main?.pressure}} mBar</p>
    <p>Humidity: {{(weatherCurrent$ | async)?.main?.humidity}} %</p>
  </mat-card-content>
</mat-card>
