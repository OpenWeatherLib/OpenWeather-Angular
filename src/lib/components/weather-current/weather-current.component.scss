@import "../../../styles/ga-common.scss";
