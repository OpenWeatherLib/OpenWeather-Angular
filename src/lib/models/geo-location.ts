import { classJsonKey } from "../decorator";

@classJsonKey("", "")
export class GeoLocation {
  lat: number = 720.0;
  long: number = 720.0;

  isDefault(): boolean {
    return this.lat === 720.0 && this.long === 720.0;
  }
}
