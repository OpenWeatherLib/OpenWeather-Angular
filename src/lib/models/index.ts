export * from "./address-component";
export * from "./city";
export * from "./city2";
export * from "./clouds";
export * from "./coordinates";
export * from "./coordinates2";
export * from "./geometry";
export * from "./main";
export * from "./rain";
export * from "./sys";
export * from "./weather-current";
export * from "./weather-forecast";
export * from "./weather-forecast-part";
export * from "./weather-part";
export * from "./wind";
export * from "./uv-index";
