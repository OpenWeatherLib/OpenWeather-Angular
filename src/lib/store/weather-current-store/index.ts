export * from "./weather-current.actions";
export * from "./weather-current.effects";
export * from "./weather-current.reducer";
export * from "./weather-current.selectors";
export * from "./weather-current.state";
