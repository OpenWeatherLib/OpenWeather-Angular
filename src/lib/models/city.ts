import { GeoLocation } from "./geo-location";
import { classJsonKey } from "../decorator";

@classJsonKey("", "")
export class City {
  id: number = 0;
  name: string = "";
  country: string = "";
  population: number = 0;
  geoLocation: GeoLocation = new GeoLocation();

  isDefault(): boolean {
    return this.id === 0 && this.name === "" && this.country === "" && this.population === 0 && this.geoLocation.isDefault();
  }
}
