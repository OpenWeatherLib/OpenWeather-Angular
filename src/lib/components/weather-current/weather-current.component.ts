import { Component, OnInit } from "@angular/core";

import { BaseComponent } from "@lib/components/base-component/base.component";
import { ApiCallState } from "@lib/enums";
import { WeatherCurrent } from "@lib/models";
import { OpenWeatherService } from "@lib/services";

@Component({
  selector: "ga-weather-current",
  templateUrl: "./weather-current.component.html",
  styleUrls: ["./weather-current.component.scss"]
})
export class WeatherCurrentComponent extends BaseComponent implements OnInit {

  currentWeather: WeatherCurrent = null;

  constructor(private readonly openWeatherService: OpenWeatherService) {
    super();
  }

  ngOnInit() {
    this.registerSubscription(
      this.openWeatherService.currentWeather()
        .subscribe(currentWeather => {
          if (currentWeather) {
            this.currentWeather = currentWeather;
          }
        }));
  }
}
