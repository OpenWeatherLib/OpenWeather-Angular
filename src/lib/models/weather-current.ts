import "@lib/extensions/string.extensions";
import { classJsonKey } from "@lib/decorator";
import WeatherCondition from "@lib/enums/weather-condition.enum";
import { City } from "./city";

@classJsonKey(String().empty, "sys")
export class WeatherCurrent {
    icon: string = String().empty;
    description: string = String().empty;
    weatherCondition: WeatherCondition = WeatherCondition.null;
    temperature: number = 0.0;
    temperatureMin: number = 0.0;
    temperatureMax: number = 0.0;
    humidity: number = 0.0;
    pressure: number = 0.0;
    visibility: number = 0.0;
    cloudsAll: number = 0.0;
    windSpeed: number = 0.0;
    windDegree: number = 0.0;
    dateTime: Date = new Date();
    sunriseTime: Date = new Date();
    sunsetTime: Date = new Date();
    city: City = new City();
    lastUpdate: Date = new Date();

    isDefault(): boolean {
        return this.icon.isNullOrEmpty()
            && this.description.isNullOrEmpty()
            && this.weatherCondition === WeatherCondition.null
            && this.temperature === 0.0
            && this.temperatureMin === 0.0
            && this.temperatureMax === 0.0
            && this.humidity === 0.0
            && this.pressure === 0.0
            && this.visibility === 0.0
            && this.cloudsAll === 0.0
            && this.windSpeed === 0.0
            && this.windDegree === 0.0
            && this.city.isDefault();
    }
}
