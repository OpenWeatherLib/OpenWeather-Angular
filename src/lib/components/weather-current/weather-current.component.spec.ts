import { NO_ERRORS_SCHEMA } from "@angular/core";
import { async, ComponentFixture, TestBed } from "@angular/core/testing";
import { combineReducers, Store, StoreModule } from "@ngrx/store";
import { of } from "rxjs";

import { WeatherCurrent } from "@lib/models";
import { weatherCurrentReducer } from "@lib/store/weather-current-store";
import { RootState } from "@lib/store/root-state";
import { WeatherCurrentComponent } from "./weather-current.component";

describe("WeatherCurrentComponent", () => {
  let component: WeatherCurrentComponent;
  let store$: Store<RootState>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [
        WeatherCurrentComponent
      ],
      imports: [
        StoreModule.forRoot({ weatherCurrent: combineReducers(weatherCurrentReducer) })
      ],
      schemas: [NO_ERRORS_SCHEMA]
    }).compileComponents();

    store$ = TestBed.get(Store);
    store$.select = jest.fn()
      .mockReturnValueOnce(of(false))
      .mockReturnValueOnce(of({
        coord: undefined,
        weather: [],
        base: "",
        main: undefined,
        visibility: 10,
        wind: undefined,
        clouds: undefined,
        dt: 0,
        sys: undefined,
        id: 0,
        name: "Name",
        cod: 0,
        weatherCondition: undefined
      }));

    const fixture: ComponentFixture<WeatherCurrentComponent> = TestBed.createComponent(WeatherCurrentComponent);
    component = fixture.debugElement.componentInstance;
  });

  test("should create the app", async(() => {
    // Assert
    expect(component).toBeTruthy();
  }));

  describe("ngOnInit", () => {
    test("isLoading$", done => {
      // Arrange 
      component.ngOnInit();

      // Act
      component.isLoading$
        .subscribe({
          next: (isLoading: boolean) => {
            // Assert
            expect(isLoading).toBeFalsy();
          },
          complete: () => done()
        });
    });

    test("weatherCurrent$", done => {
      // Arrange 
      component.ngOnInit();

      // Act
      component.weatherCurrent$
        .subscribe({
          next: (weatherCurrent: WeatherCurrent) => {
            // Assert
            expect(weatherCurrent).toMatchSnapshot({
              coord: undefined,
              weather: [],
              base: "",
              main: undefined,
              visibility: 10,
              wind: undefined,
              clouds: undefined,
              dt: 0,
              sys: undefined,
              id: 0,
              name: "Name",
              cod: 0,
              weatherCondition: undefined
            });
          },
          complete: () => done()
        });
    });
  });
});
