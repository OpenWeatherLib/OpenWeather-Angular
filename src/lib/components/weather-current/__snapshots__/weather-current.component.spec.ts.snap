// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCurrentComponent ngOnInit weatherCurrent$ 1`] = `
Object {
  "base": "",
  "clouds": undefined,
  "cod": 0,
  "coord": undefined,
  "dt": 0,
  "id": 0,
  "main": undefined,
  "name": "Name",
  "sys": undefined,
  "visibility": 10,
  "weather": Array [],
  "weatherCondition": undefined,
  "wind": undefined,
}
`;
